src/client.rs

```diff
@@ -435,8 +430,7 @@ impl Client {
             RegistryOperation::Push => format!("repository:{}:pull,push", image.repository()),
         };
 
-        let challenge = &challenge_opt[0];
-        let realm = challenge.realm.as_ref().unwrap();
+        let realm = challenge.realm.as_ref();
```

Comment:
Do we have a mem leak from the `Box`ed realm now?

Reply:
I don't think so... please correct me if I'm wrong.

`realm` holds a reference to a `&str` that is built starting from a value provided by a `ChallengeParser`. `ChallengeParser` has a lifetime equal to the value it's built against: the `HeaderValue`: https://github.com/krustlet/oci-distribution/blob/a7189f495aa9efebda878b95121f7e2981dee095/src/client.rs#L1375-L1379

The `HeaderValue` is created inside of the `auth` function, https://github.com/krustlet/oci-distribution/blob/a7189f495aa9efebda878b95121f7e2981dee095/src/client.rs#L407-L410

I think `realm` and `HeaderValue`all go away at the same time, once the scope of `auth` is left.

Am I missing something?

Reply:
Not at all, all fine! Got confused with removed code and thought we were doing an `into_raw()` instead of `into_boxed_str()` for its creation.